webapp/vite.config.ts

```diff
@@ -1,29 +1,26 @@
-import { defineConfig } from "vite";
+import { type ProxyOptions, defineConfig } from "vite";
 import react from "@vitejs/plugin-react-swc";
 
 const SERVER_URL = "http://localhost:8080";
 
+const proxyOptions: ProxyOptions = {
+  target: SERVER_URL,
+  changeOrigin: true, // Recommended for avoiding CORS issues
+  ws: true, // WebSocket support for hot module replacement
+};
+
 // https://vitejs.dev/config/
 export default defineConfig(({ mode }) => {
   const isDesktopMode = mode === "desktop";
 
   return {
-    // When the web application is running in Desktop mode,
-    // the web app is served at the `/static` entry point
-    base: isDesktopMode ? "/static/" : "/",
+    base: isDesktopMode ? "/static/" : "/", // Sets base to "/static/" in Desktop mode for compatible static file serving, otherwise root.
     plugins: [react({ devTarget: "es2022" })],
     server: {
       port: 3000,
       strictPort: true,
       proxy: {
-        // Main API URLs
-        "/v1": SERVER_URL,
-        // Core API URLs
-        "/health": SERVER_URL,
-        "/kill": SERVER_URL,
-        "/version": SERVER_URL,
-        // WebSocket
-        "/ws": SERVER_URL,
+        "^/(v1|health|kill|version|ws|openapi.json)": proxyOptions,
```

Comment:
Pour info, à part les routes qui commencent par "/v1", il faut gérer les routes suivantes : ['/', '/health', '/kill', '/openapi.json', '/redoc', '/version', '/ws']

Reply:
Il manque donc "/redoc" (je ne sais pas à quoi ça correspond).

Reply:
=>  HTML response that loads ReDoc for the alternative API docs (normally served at /redoc).

Reply:
Il a mis une regex qui les regroupe tous (je ne savais pas au'on pouvais faire ça).
Il manque juste `redoc` du coup.